src/types/models.ts

```diff
@@ -46,27 +46,62 @@ export type ThermostatDeviceType = typeof THERMOSTAT_DEVICE_TYPES[number]
 
 export type HvacModeSetting = "off" | "heat" | "cool" | "heat_cool"
 
-export interface ThermostatDeviceProperties extends CommonDeviceProperties {
-  is_cooling: boolean
+export type ClimateSetting = {
+  automatic_heating_enabled?: boolean
+  automatic_cooling_enabled?: boolean
+  hvac_mode_setting?: HvacModeSetting
+  cooling_set_point_celsius?: number
+  heating_set_point_celsius?: number
+  cooling_set_point_fahrenheit?: number
+  heating_set_point_fahrenheit?: number
+  manual_override_allowed: boolean
+}
+export interface BaseThermostatDeviceProperties extends CommonDeviceProperties {
+  temperature_fahrenheit: number
+  temperature_celsius: number
+  relative_humidity: number // [0,1]
+  can_enable_automatic_heating: boolean
+  can_enable_automatic_cooling: boolean
+  available_hvac_mode_settings: Array<HvacModeSetting>
   is_heating: boolean
+  is_cooling: boolean
   is_fan_running: boolean
-  has_direct_power: boolean
-  relative_humidity: number
-  temperature_celsius: number
-  temperature_fahrenheit: number
-  current_climate_setting: {
-    hvac_mode_setting: HvacModeSetting
-    manual_override_allowed: boolean
-    automatic_cooling_enabled: boolean
-    automatic_heating_enabled: boolean
-    cooling_set_point_celsius: number
-    cooling_set_point_fahrenheit: number
+  is_temporary_manual_override_active: boolean // this is true if the current thermostat settings differ that what is on seam, and `current_climate_setting.manual_override_allowed: true`
+  current_climate_setting: ClimateSetting // this field has the actual climate setting as it is on the device
+  default_climate_setting?: ClimateSetting
+  is_climate_setting_schedule_active: boolean
+  active_climate_setting_schedule?: ClimateSettingSchedule
+  is_cooling_available: false
+  is_heating_available: false
+}
+
+export type CoolingThermostatProperties = BaseThermostatDeviceProperties & {
+  is_cooling_available: true
```

Comment:
This is for the discriminated union to work. I worry it may look weird in the docs.

Reply:
That's ok